service/src/main/kotlin/fi/espoo/evaka/shared/job/ScheduledJobs.kt

```diff
@@ -247,18 +247,16 @@ WITH attendances_to_end AS (
     SELECT ca.id
     FROM child_attendance ca
     JOIN daycare u ON u.id = ca.unit_id
+    LEFT JOIN placement p ON p.child_id = ca.child_id AND daterange(p.start_date, p.end_date, '[]') @> ${bind(today)}
+    LEFT JOIN service_need sn ON sn.placement_id = p.id AND daterange(sn.start_date, sn.end_date, '[]') @> ${bind(today)}
+    LEFT JOIN backup_care bc ON ca.child_id = bc.child_id AND daterange(bc.start_date, bc.end_date, '[]') @> ${bind(today)}
     WHERE ca.end_time IS NULL AND (
         -- Unit is not open through midnight (checking until midnight but in practice no unit closes at midnight)
         u.shift_care_operation_times[extract(isodow FROM ca.date)].end IS DISTINCT FROM '23:59'::time OR
-        -- No placement to this unit anymore, as of today
-        NOT EXISTS (
-            SELECT 1 FROM placement p
-            LEFT JOIN backup_care bc ON ca.child_id = bc.child_id AND ${bind(today)} BETWEEN bc.start_date AND bc.end_date
-            WHERE
-                p.child_id = ca.child_id AND
-                (p.unit_id = ca.unit_id OR bc.unit_id = ca.unit_id) AND
-                ${bind(today)} BETWEEN p.start_date AND p.end_date
-        )
+        -- Child does not have shift care in service need
+        sn.shift_care IS DISTINCT FROM 'NONE' OR
+        -- No (backup) placement to this unit anymore, as of today
+        (p.unit_id IS DISTINCT FROM u.id AND bc.unit_id IS DISTINCT FROM u.id)
```

Comment:
`NOT EXISTS`-sääntöjen lisääminen olisi parempi kuin `LEFT JOIN`:ien lisääminen, koska joinit voi aiheuttaa sen että `SELECT`-kysely tuottaa useita rivejä. Ei siinä ehkä tässä kohtaa ole suurta haittaa, mutta kuitenkin.

Reply:
... tai itse asiassa voi tosta bugikin tulla. Pitäisikö ainakin join-ehtoihin lisätä suoraan toi `p.unit_id = u.id` / `bc.unit_id = u.id`

Reply:
Mun nähdäkseni tässä ei voi koska kaikissa on join-ehtona yksittäinen päivä

Reply:
ja p.unit_id = u.id joinissa ei toimi koska toiseenkin yksikköön oleva sijoitus tarvitaan koska se vaaditaan joinissa palveluntarpeeseen

Reply:
Totta 👍